client/src/app/components/schema-defined-fields/SchemaAsFields.tsx

```diff
@@ -0,0 +1,238 @@
+/* eslint-disable no-case-declarations */
+import React from "react";
+import { useTranslation } from "react-i18next";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as yup from "yup";
+
+import {
+  ActionGroup,
+  Button,
+  ButtonVariant,
+  Card,
+  CardBody,
+  CardTitle,
+  Checkbox,
+  Form,
+  Title,
+} from "@patternfly/react-core";
+
+import {
+  HookFormPFGroupController,
+  HookFormPFTextInput,
+} from "@app/components/HookFormPFFields";
+import { JsonSchemaObject } from "@app/api/models";
+
+export interface JsonSchemaValues {
+  [key: string]: string | string[] | boolean | JsonSchemaValues | undefined;
+}
+
+export interface SchemaAsFieldsProps {
+  jsonDocument: object;
+  jsonSchema: JsonSchemaObject;
+  onDocumentSaved?: (newJsonDocument: object) => void;
+}
+
+export const SchemaAsFields: React.FC<SchemaAsFieldsProps> = ({
+  jsonDocument,
+  jsonSchema,
+  onDocumentSaved,
+}) => {
+  const { t } = useTranslation();
+
+  const jsonSchemaToYupSchema = (
```

Comment:
This should be a util function that can have some unit tests.  We'd want a few unit tests that can prove...

Given a simple schema (one property in an object), turn it into a yup validator and then run validation against a good and a bad sample.  That will let us be reasonable sure that what we say can work will work w/o having to manually test everything.

All that being said, looks like decent coverage.

There is always something like https://www.npmjs.com/package/json-schema-yup-transformer that could be adopted if more exotic use cases come up.